lib/op.ml

```diff
@@ -136,8 +143,8 @@ type 'aop op1 =
   | Promote (* Promote to higher precision *)
   | Demote  (* Demote to lower precision *)
   | ArchOp1 of 'aop
-
-
+  | CheckCanonical (* Check that the PAC field of a virtual address is canonical *)
```

Comment:
Perhaps we could consider having those operations in `AArch64Op.ml` as they are arch specific. Notice that this is going halfway, as, for instance, there are no arch-specific faults for instance.

Nevertheless, as the mechanism for arch-specific operations is here, let us use it :)

Reply:
I agree with @maranget here